integration_tests/import_test.py

```diff
@@ -0,0 +1,118 @@
+import os
+import re
+import subprocess
+
+from keras import backend
+
+BACKEND_REQ = {
+    "tensorflow": "tensorflow",
+    "torch": "torch torchvision",
+    "jax": "jax jaxlib",
+}
+
+
+def setup_package():
+    subprocess.run("rm -rf tmp_build_dir", shell=True)
+    build_process = subprocess.run(
+        "python3 pip_build.py",
+        capture_output=True,
+        text=True,
+        shell=True,
+    )
+    print(build_process.stdout)
+    match = re.search(
+        r"\s[^\s]*\.whl",
+        build_process.stdout,
+    )
+    if not match:
+        raise ValueError("Installing Keras package unsuccessful. ")
+        print(build_process.stderr)
+    whl_path = match.group()
+    return whl_path
+
+
+def create_virtualenv():
+    env_setup = [
+        # Create and activate virtual environment
+        "python3 -m venv test_env",
+        # "source ./test_env/bin/activate",
+    ]
+    os.environ["PATH"] = (
+        "/test_env/bin/" + os.pathsep + os.environ.get("PATH", "")
+    )
+    run_commands_local(env_setup)
+
+
+def manage_venv_installs(whl_path):
+    other_backends = list(set(BACKEND_REQ.keys()) - {backend.backend()})
+    install_setup = [
+        # Installs the backend's package and common requirements
+        "pip install " + BACKEND_REQ[backend.backend()],
+        "pip install -r requirements-common.txt",
+        "pip install pytest",
+        # Ensure other backends are uninstalled
+        "pip uninstall -y "
+        + BACKEND_REQ[other_backends[0]]
+        + " "
+        + BACKEND_REQ[other_backends[1]],
+        # Install `.whl` package
+        "pip install " + whl_path + " --force-reinstall --no-dependencies",
+    ]
+    run_commands_venv(install_setup)
+
+
+def run_keras_flow():
+    test_script = [
+        # Runs the example script
+        "python -m pytest integration_tests/basic_full_flow.py",
```

Comment:
How do we check it actually works? I think we need some kind of output canary to check the script has successfully completed

Reply:
Actually `subprocess.run(command)` prints the output of the command directly to the console, so no output canary is needed. Here's the console output, for example:

![image](https://github.com/keras-team/keras/assets/60985914/fc39aaa6-b703-47fa-9ed4-7f4b13afcc93)

Reply:
And here's an example output when it failed (because I added `import jax` in saving_lib.py):

![image](https://github.com/keras-team/keras/assets/60985914/55063889-05f3-4933-9329-7153a9acadac)